Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -2344,7 +2344,8 @@ public ReferenceDescriptionCollection FetchReferences(NodeId nodeId)
             bool requireEncryption = securityPolicyUri != SecurityPolicies.None;
             if (!requireEncryption)
             {
-                requireEncryption = identityPolicy.SecurityPolicyUri != SecurityPolicies.None;
+                if (identityPolicy.SecurityPolicyUri != null)
+                    requireEncryption = identityPolicy.SecurityPolicyUri != SecurityPolicies.None;
             }
 
```

Comment:
You need to check also for empty string, like in the version from _UA-.NET-Legacy_ repo:
```
            bool requireEncryption = securityPolicyUri != SecurityPolicies.None;

            if (!requireEncryption)
            {
                requireEncryption = identityPolicy.SecurityPolicyUri != SecurityPolicies.None && 
                    !String.IsNullOrEmpty(identityPolicy.SecurityPolicyUri);
            }           
```